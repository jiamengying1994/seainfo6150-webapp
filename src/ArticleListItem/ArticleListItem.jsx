import React from "react";
import ArticleTextToggleButton from "../ArticleTextToggleButton/ArticleTextToggleButton";
import styles from "./ArticleListItem.module.css";

export class ArticleListItem extends React.Component {
  constructor(props) {
      super(props);
      this.state = {isToggleOn: false};
      this.handleClick = this.handleClick.bind(this);
   }

  handleClick() {
       this.setState(state => ({
         isToggleOn: !this.state.isToggleOn
       }));
  }

  render() {
  return (
    <>
      <div>
        <h2 className={styles.title}>{this.props.articleListItem.title}</h2>
        <p className={styles.date} key={this.props.articleListItem.title}>
         {this.state.isToggleOn?this.props.articleListItem.displayDate: ""}
        </p>
        {this.state.isToggleOn?this.props.articleListItem.shortText: ""}
        <p key={this.props.articleListItem.slug}>
        </p>
      </div>
      <ArticleTextToggleButton
      onClick={this.handleClick}
      buttonText={this.state.isToggleOn?'Show less':'Show more'}/>
    </>
  );
  }
}


export default ArticleListItem;