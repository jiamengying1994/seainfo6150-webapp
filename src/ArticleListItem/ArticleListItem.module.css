.title {

color: red;
font-size: 20px;
margin-bottom: 40px;
}

.date{
font-weight: bold;
}