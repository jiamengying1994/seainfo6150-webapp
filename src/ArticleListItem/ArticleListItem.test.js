import React from "react";
import ArticleListItem from "./ArticleListItem";
import { fireEvent } from "@testing-library/react";

describe("ArticleListItem tests", () => {


  it("renders correctly after button is clicked", () => {
    const al = {"title": "hello", "image": "https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"}
    const { container, getByText } = render(
      <ArticleListItem articleListItem={al} />
    );
    const button = getByText("Show more");
    fireEvent.click(button);
    expect(container).toMatchSnapshot();
  });
});