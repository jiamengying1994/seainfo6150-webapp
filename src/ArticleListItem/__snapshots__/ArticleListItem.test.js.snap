// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItem tests renders correctly after button is clicked 1`] = `
<div>
  <div>
    <div
      class="article_image"
    >
      <img
        alt="hello"
      />
    </div>
    <h2
      class="title"
    >
      hello
    </h2>
    <p
      class="date"
    />
    <p />
  </div>
  <button
    class="button"
  >
    Show less
  </button>
</div>
`;
